src/Chainweb/Version.hs

```diff
@@ -615,14 +615,31 @@ vuln797FixDate Testnet04 = epoch
 vuln797FixDate Mainnet01 = [timeMicrosQQ| 2019-12-10T21:00:00.0 |]
 {-# INLINE vuln797FixDate #-}
 
--- | Upgrade coin v2 at time, or at block height 1
--- | Must be BEFORE 'txEnabledDate' in mainnet.
-upgradeCoinV2Date :: ChainwebVersion -> Maybe (Time Micros)
-upgradeCoinV2Date Test{} = Nothing
-upgradeCoinV2Date TimedConsensus{} = Nothing
-upgradeCoinV2Date PowConsensus{} = Nothing
-upgradeCoinV2Date TimedCPM{} = Nothing
-upgradeCoinV2Date FastTimedCPM{} = Nothing
-upgradeCoinV2Date Development = Just [timeMicrosQQ| 2019-12-14T18:50:00.0 |]
-upgradeCoinV2Date Testnet04 = Nothing
-upgradeCoinV2Date Mainnet01 = Just [timeMicrosQQ| 2019-12-17T15:00:00.0 |]
+-- | Mainnet upgrade to coin v2 at time at @[timeMicrosQQ| 2019-12-17T15:00:00.0 |]@,
+-- which was BEFORE 'txEnableDate'. All other chainweb versions use coin v2 from
+-- the genesis block on.
+--
+-- This function provides the block heights when coin v2 became effective on the
+-- respective chains.
+--
+coinV2Upgrade
+    :: (Ord a, Num a)
+    => ChainwebVersion
+    -> ChainId
+    -> a
+        -- ^ FIXME this should be BlockHeight
```

Comment:
I have another draft PR that moves `BlockHeight` into a module on its own, which is a prerequisite for this.